packages/cli/openapi-ir-to-fern/src/converters/convertEndpoint.ts

```diff
@@ -226,13 +229,128 @@ export function convertEndpoint({
 
+    if (endpoint.examples.length > 0) {
+        convertedEndpoint.examples = convertEndpointExamples(endpoint.examples);
+    }
+
     return {
         value: convertedEndpoint,
         schemaIdsToExclude,
         additionalTypeDeclarations,
     };
 }
 
+interface NamedFullExample {
+    name: string;
+    value: FullExample;
+}
+
+function convertEndpointExamples(endpointExamples: EndpointExample[]): RawSchemas.ExampleEndpointCallSchema[] {
+    return endpointExamples.map((endpointExample) => {
+        return convertEndpointExample(endpointExample);
+    });
+}
+
+function convertEndpointExample(endpointExample: EndpointExample): RawSchemas.ExampleEndpointCallSchema {
+    return {
+        "path-parameters":
+            endpointExample.pathParameters != null
+                ? convertNamedFullExamplesToExampleTypeReferenceSchemas(endpointExample.pathParameters)
+                : undefined,
+        "query-parameters":
+            endpointExample.queryParameters != null
+                ? convertNamedFullExamplesToExampleTypeReferenceSchemas(endpointExample.queryParameters)
+                : undefined,
+        headers:
+            endpointExample.headers != null
+                ? convertNamedFullExamplesToExampleTypeReferenceSchemas(endpointExample.headers)
+                : undefined,
+        request:
+            endpointExample.request != null
+                ? convertFullExampleToExampleTypeReferenceSchema(endpointExample.request)
+                : undefined,
+        response:
+            endpointExample.response != null
+                ? { body: convertFullExampleToExampleTypeReferenceSchema(endpointExample.response) }
+                : undefined,
+    };
+}
+
+function convertNamedFullExamplesToExampleTypeReferenceSchemas(
+    namedFullExamples: NamedFullExample[]
+): Record<string, ExampleTypeReferenceSchema> {
+    const result: Record<string, ExampleTypeReferenceSchema> = {};
+    namedFullExamples.map(
+        (namedFullExample) =>
+            (result[namedFullExample.name] = convertFullExampleToExampleTypeReferenceSchema(namedFullExample.value))
+    );
+    return result;
+}
+
+function convertFullExampleToExampleTypeReferenceSchema(fullExample: FullExample): ExampleTypeReferenceSchema {
+    return FullExample._visit<ExampleTypeReferenceSchema>(fullExample, {
+        primitive: (primitive) => convertPrimitiveExampleToExampleTypeReferenceSchema(primitive),
+        object: (object) => convertFullExamplePropertiesToExampleTypeReferenceSchema(object.properties),
+        array: (array) => convertFullArrayExampleToExampleTypeReferenceSchema(array),
+        map: (map) => convertFullMapExampleToExampleTypeReferenceSchema(map),
+        oneOf: (oneOf) => convertFullOneOfExampleToExampleTypeReferenceSchema(oneOf),
+        enum: (value) => value,
+        literal: (value) => value,
+        unknown: (unknown) => convertFullExampleToExampleTypeReferenceSchema(unknown),
+        _unknown: () => "unknown",
```

Comment:
@dsinghvi would you prefer that I not use the visitor and go with the `if/else` switch and throw an exception in these cases? Or is there another default value we should go with? I don't think we can assume the parsed OpenAPI result is an unknown object, i.e. `{}`, can we? This probably won't even be reachable anyways.

Reply:
yeah would be okay for you to throw an exception here

Reply:
the `unknown` case shouldnt matter here, nothing needs to be "forward compat"